Formula/cargo-zigbuild.rb

```diff
@@ -0,0 +1,30 @@
+class CargoZigbuild < Formula
+  desc "Compile Cargo project with zig as linker"
+  homepage "https://github.com/messense/cargo-zigbuild"
+  url "https://github.com/messense/cargo-zigbuild/archive/refs/tags/v0.12.2.tar.gz"
+  sha256 "90397c5170eb02ebcf8c378e499f2c9be3b1b63c85d20e1e0584aec459321ed7"
+  license "MIT"
+  head "https://github.com/messense/cargo-zigbuild.git", branch: "main"
+
+  depends_on "rustup-init" => :test
+  depends_on "rust"
+  depends_on "zig"
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    # https://github.com/ziglang/zig/issues/10377
+    ENV.delete "CPATH"
+
+    system "#{Formula["rustup-init"].bin}/rustup-init", "-y", "--no-modify-path"
+    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
+    system "rustup", "target", "add", "aarch64-unknown-linux-gnu"
```

Comment:
Can the `rust` formula emit object code only for native targets? I didn't think we configured `rust` for that. Or is it the default?

Reply:
I'm not sure what you mean. I'm testing cross compilation because it better represents the use case of `cargo-zigbuild`.

Reply:
Well, what I mean is:

> I'm not sure what you mean. I'm testing cross compilation because it better represents the use case of `cargo-zigbuild`.

Do we really need `rustup-init` for this, or can we manage with just the `rust` formula?

Reply:
Only `rustup` can install addition targets, the `rust` formula can not.

Reply:
That's not my question though -- whether the `rust` formula's `rustc` can generate object code for a non-native target is a property of the configuration of the LLVM backend, and not whether it includes `rustup` or not. See https://rustc-dev-guide.rust-lang.org/backend/codegen.html.

Does Rust configure LLVM to build only the native target? Note that a default build of LLVM includes all targets by default, so unless Rust specifies the targets to build when building LLVM, then it likely can generate object code for other targets too.

Reply:
> Does Rust configure LLVM to build only the native target?

I think not, it can generate object code for other targets. Here we just need `rustup` to install the Rust standard library for the other targets.

Reply:
I decided to check this for myself, and it appears to me that we don't need `rustup` to emit object code for a non-native target:
```
❯ /opt/homebrew/bin/rustc --print=target-list | rg x86_64
x86_64-apple-darwin
x86_64-apple-ios
x86_64-apple-ios-macabi
x86_64-apple-tvos
x86_64-apple-watchos-sim
x86_64-fortanix-unknown-sgx
x86_64-fuchsia
x86_64-linux-android
x86_64-pc-solaris
x86_64-pc-windows-gnu
x86_64-pc-windows-gnullvm
x86_64-pc-windows-msvc
x86_64-sun-solaris
x86_64-unknown-dragonfly
x86_64-unknown-freebsd
x86_64-unknown-haiku
x86_64-unknown-hermit
x86_64-unknown-illumos
x86_64-unknown-l4re-uclibc
x86_64-unknown-linux-gnu
x86_64-unknown-linux-gnux32
x86_64-unknown-linux-musl
x86_64-unknown-netbsd
x86_64-unknown-none
x86_64-unknown-none-linuxkernel
x86_64-unknown-openbsd
x86_64-unknown-redox
x86_64-unknown-uefi
x86_64-uwp-windows-gnu
x86_64-uwp-windows-msvc
x86_64-wrs-vxworks
```
The linker consumes only object files, so it seems to me that we should be able to do this test without requiring an installation of `rustup-init`.

Reply:
Well, we need to test the linker so it's better to build an executable.

```bash
$ cat src/main.rs
fn main() {
    println!("Hello, world!");
}
$ /opt/homebrew/bin/rustc --target aarch64-unknown-linux-gnu src/main.rs
error[E0463]: can't find crate for `std`
  |
  = note: the `aarch64-unknown-linux-gnu` target may not be installed
  = help: consider downloading the target with `rustup target add aarch64-unknown-linux-gnu`

error: cannot find macro `println` in this scope
 --> src/main.rs:2:5
  |
2 |     println!("Hello, world!");
  |     ^^^^^^^

error: requires `sized` lang_item

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0463`.
$ fn main() {
}
$ /opt/homebrew/bin/rustc --target aarch64-unknown-linux-gnu src/main.rs
error[E0463]: can't find crate for `std`
  |
  = note: the `aarch64-unknown-linux-gnu` target may not be installed
  = help: consider downloading the target with `rustup target add aarch64-unknown-linux-gnu`

error: requires `sized` lang_item

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0463`.
```

Reply:
Even an empty program requires the stdlib? Odd.

Ok, thanks for the explanation!